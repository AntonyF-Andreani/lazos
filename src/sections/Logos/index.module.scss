.root {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  overflow: hidden;
  padding: 15px;
  max-width: 1200px;
  margin: auto;

  @media only screen and (max-width: 768px) {
    padding: 0;
  }
}

.title {
  letter-spacing: -0.05em;
  font-weight: 700;
  font-size: 2.3em;
  color: #212529;
  margin: 20px;
  margin-bottom: 40px;

  @media only screen and (max-width: 768px) {
    font-size: 2em;
  }

  @media only screen and (max-width: 425px) {
    margin-bottom: 20px;
    font-size: 1.6em;
  }

  @media only screen and (max-width: 375px) {
    font-size: 1.2em;
  }
}

.slide {
  width: 390px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.image {
  object-fit: cover;
  object-position: center;
  height: 80px;
  background: transparent;
  background-color: transparent;

  &[data-isactive="false"] {
    opacity: 0.5;
  }
}

.selectDisable {
  user-select: none;
}

.container {
  margin: 15px;
  width: 100%;
}
