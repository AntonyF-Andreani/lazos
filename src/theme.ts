import { createTheme } from "@mui/material/styles";

const theme = createTheme({
  palette: {
    primary: {
      main: "#fa1a22", //mas opaco
      dark: "#b6040b", //mas claro
      light: "#D71920",
      //D71920
    },
    secondary: {
      main: "#212529",
      dark: "#32383e",
    },
    text: {
      primary: "#212529",
      secondary: "#777",
    },
  },
  typography: {
    fontFamily: "'Work Sans', sans-serif",
  },
  components: {
    MuiDivider: {
      styleOverrides: {
        root: {
          backgroundColor: "rgba(0, 0, 0, 0.06)",
        },
      },
    },
  },
});

export default theme;
