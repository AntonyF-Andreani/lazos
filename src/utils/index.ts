import { ImageLoaderProps } from "next/image";

/**
 * Utilidad para extraerla URL con formatos optimizados para las imagenes del CMS (Servicio de Imagenes)
 * https://www.storyblok.com/docs/image-service
 */
export const StoryBlockImageURL = ({
  src,
  width,
  quality = 20,
  format = "webp",
  height = 0,
}: StoryBlockImageURLParams) => {
  return `${src}/m/${width}x${height}/filters:quality(${quality}):format(${format})`;
};

interface StoryBlockImageURLParams {
  /** URL de la imagen como la provee Storyblok */
  src: string;
  /** Width of image. The Storyblok API infers aspect ratio */
  width: number;
  /** Quality of image, for default its 20 (20%). */
  quality?: number;
  /** Format of image, for default its webp */
  format?: "webp" | "jpeg" | "png";
  /** Height of the image */
  height?: number;
}

export const StoryBlockLoader = ({ src, width, quality = 20 }: ImageLoaderProps) => {
  return StoryBlockImageURL({ src, width, quality });
};

/**
 * Util for join classes undefined or string and filter undefined
 * @param args classNames
 *
 *  @example const className = joinClasses("a", "b", undefined, "c") // "a b c"
 */
export const joinClasses = (...args: (string | undefined)[]): string => {
  return args.filter((x) => x).join(" "); // undefined is falsey
};

/**
 * Util to test diferrent viewports, must be used with jest.
 *
 * This util make a mock implementation of window.matchMedia
 *
 * @example prepareViewport("(max-width: 768px)")
 */
export const prepareViewport = (mediaQuery: string) => {
  Object.defineProperty(window, "watchMedia", {
    writable: true,
    configurable: true,
    value: mediaQuery,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    return {
      matches: query === mediaQuery,
      media: "",
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};
