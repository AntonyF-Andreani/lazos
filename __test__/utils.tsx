import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";

import theme from "theme";

/**
 * Colocar todos los providers, se puede mockear redux y las funciones
 */
const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement) => render(ui, { wrapper: AllTheProviders });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
