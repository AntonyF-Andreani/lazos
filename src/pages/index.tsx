import React from "react";
import type { GetStaticProps, InferGetStaticPropsType } from "next";

import Informative from "sections/Informative";
import { Content } from "services/Cms/types";
import { CmsService } from "services/Cms";
import Main from "layout/Main";
import Logos from "sections/Logos";
import MainBanner from "sections/MainBanner";
import Steps from "sections/Steps";
import InfoSection from "sections/Info";
import Testimonials from "sections/Testimonials";
import { logger } from "utils";

export const getStaticProps: GetStaticProps<Content> = async () => {
  const { data } = await CmsService.getContentFetch();

  const {
    data: {
      LazoscontentItem: { content },
    },
  } = data;

  logger("Home Generated")

  return {
    props: content,
    revalidate: 300, //5min
  };
};

const Home = (props: InferGetStaticPropsType<typeof getStaticProps>) => {
  const {
    sidebar,
    logo_navbar,
    logo_navbar_dark,
    banner_image,
    title_logos,
    logos,
    program_information_title,
    program_information_description,
    program_information_title_w,
    program_information_description_w,
    title_steps,
    steps,
    information_section: [{ link, description, button_text }],
    media,
    logo_fundacion,
    logo_andreani,
    title_testimonials,
    testimonials_image,
    testimonials,
    terms,
  } = props;

  return (
    <Main
      darkImage={logo_navbar_dark}
      defaultImage={logo_navbar}
      logo_andreani={logo_andreani}
      logo_fundacion={logo_fundacion}
      media={media ? media[0] : undefined}
      sidebarData={sidebar}
    >
      <MainBanner image={banner_image} />
      <InfoSection button_link={link.url} button_text={button_text} text={description} />
      <Steps data={steps} terms={terms} title={title_steps} />
      <Informative
        description={program_information_description}
        description_white={program_information_description_w}
        title={program_information_title}
        title_white={program_information_title_w}
      />
      <Logos data={logos} title={title_logos} />
      <Testimonials data={testimonials} image={testimonials_image} title={title_testimonials} />
    </Main>
  );
};

export default Home;
