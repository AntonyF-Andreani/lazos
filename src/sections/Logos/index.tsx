import React from "react";
import { Swiper as SwiperComponent, SwiperSlide } from "swiper/react";
import SwiperCore, { Autoplay } from "swiper";
import { Grid, Typography, useMediaQuery } from "@mui/material";
import { LazyLoadImage } from "react-lazy-load-image-component";

import styles from "./index.module.scss";

import type { ClientLogo } from "services/Cms/types";
import { joinClasses } from "utils";

export interface LogosProps {
  title: string;
  data: ClientLogo[];
}

SwiperCore.use([Autoplay]);

const Logos = ({ data, title }: LogosProps) => {
  const isTablet = useMediaQuery("(max-width: 1000px)");
  const isMobile = useMediaQuery("(max-width: 600px)");

  function getAmountOfLogos() {
    if (isMobile) {
      return 1;
    } else if (isTablet) {
      return 3;
    }

    return 5;
  }

  return (
    <Grid className={styles.root} component="section">
      <Typography className={styles.title} color="secondary" variant="h2">
        {title}
      </Typography>
      <SwiperComponent
        lazy
        loop
        autoplay={{
          delay: 3000,
          disableOnInteraction: false,
        }}
        centeredSlides={true}
        className={styles.container}
        effect="fade"
        slidesPerView={getAmountOfLogos()}
      >
        {data &&
          data.length !== 0 &&
          data.map((item, index) => {
            const { logo, _uid } = item;

            return (
              <SwiperSlide key={_uid} className={styles.slide}>
                {({ isActive }) => (
                  <LazyLoadImage
                    alt={logo.alt ? logo.alt : `Logo cliente ${index}`}
                    className={joinClasses(styles.image, styles.selectDisable)}
                    data-isactive={isActive}
                    effect="blur"
                    src={logo.filename}
                  />
                )}
              </SwiperSlide>
            );
          })}
      </SwiperComponent>
    </Grid>
  );
};

export default Logos;
