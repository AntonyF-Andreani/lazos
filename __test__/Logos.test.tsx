import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import { prepareViewport } from "../src/utils";

import Logos, { LogosProps } from "sections/Logos";

const defaultProps: LogosProps = {
  title: "test",
  data: [
    {
      _uid: "1",
      logo: {
        filename: `image-logo`,
        alt: "test",
      },
    },
    {
      _uid: "2",
      logo: {
        filename: `image-logo`,
        alt: "test",
      },
    },
    {
      _uid: "3",
      logo: {
        filename: `image-logo`,
        alt: "test",
      },
    },
  ],
};

describe("<Logos />", () => {
  beforeAll(() => {
    // Desktop viewport
    prepareViewport("(min-width:1200px)");
  });

  it("should render without crash", () => {
    const { asFragment } = render(<Logos {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have title", () => {
    const { getByText } = render(<Logos {...defaultProps} />);
    const element = getByText(defaultProps.title as string);

    expect(element).toBeInTheDocument();
  });

  it("should have the correct logo image", () => {
    const { getAllByRole } = render(<Logos {...defaultProps} />);

    const elements = getAllByRole("img");

    // loop property Swiper duplicate each logo img
    elements.map((el, index) => {
      return expect(el.getAttribute("src")).toBe(
        defaultProps.data[index]?.logo?.src ? defaultProps.data[index].logo.src : `image-logo`
      );
    });
  });

  it("should have the correct logo alt of image", () => {
    const { getAllByRole } = render(<Logos {...defaultProps} />);

    const elements = getAllByRole("img");

    // loop property Swiper duplicate each logo img
    elements.map((el, index) =>
      expect(el.getAttribute("alt")).toBe(
        defaultProps.data[index] ? defaultProps.data[index].logo.alt : `test`
      )
    );
  });

  it("should have the correct logo alt of image", () => {
    const { getAllByRole } = render(
      <Logos
        data={[
          {
            _uid: "1",
            logo: {
              filename: `image-logo`,
            },
          },
        ]}
        title={defaultProps.title}
      />
    );

    const elements = getAllByRole("img");

    // loop property Swiper duplicate each logo img
    elements.map((el) => {
      expect(el.getAttribute("alt")).toBe(`Logo cliente 0`);
    });
  });
});
