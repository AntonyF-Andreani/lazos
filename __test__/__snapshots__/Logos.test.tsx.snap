// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logos /> should render without crash 1`] = `
<DocumentFragment>
  <section
    class="MuiGrid-root root css-vj1n65-MuiGrid-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 title css-oouzf9-MuiTypography-root"
    >
      test
    </h2>
    <div
      class="swiper-container swiper-container-initialized swiper-container-horizontal container"
    >
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide swiper-slide-duplicate slide"
          data-swiper-slide-index="0"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate slide"
          data-swiper-slide-index="1"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate slide"
          data-swiper-slide-index="2"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide slide"
          data-swiper-slide-index="0"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide slide"
          data-swiper-slide-index="1"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide slide"
          data-swiper-slide-index="2"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate slide"
          data-swiper-slide-index="0"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate slide"
          data-swiper-slide-index="1"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate slide"
          data-swiper-slide-index="2"
        >
          <span
            class=" lazy-load-image-background blur"
            style="color: transparent; display: inline-block;"
          >
            <img
              alt="test"
              class="image selectDisable"
              data-isactive="false"
              src="image-logo"
            />
          </span>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
